wdl/transforms/new-base/src/main/scala/wdl/transforms/base/ast2wdlom/AstToWorkflowDefinitionElement.scala

```diff
@@ -26,26 +27,91 @@ object AstToWorkflowDefinitionElement {
                               sourceLocation: Option[SourceFileLocation],
                               bodyElements: Vector[WorkflowBodyElement]) = {
 
-    val inputsSectionValidation: ErrorOr[Option[InputsSectionElement]] = validateSize(bodyElements.filterByType[InputsSectionElement], "inputs", 1)
-    val outputsSectionValidation: ErrorOr[Option[OutputsSectionElement]] = validateSize(bodyElements.filterByType[OutputsSectionElement], "outputs", 1)
+    val inputsSectionValidation: ErrorOr[Option[InputsSectionElement]] = for {
+      inputValidateElement <- validateSize(bodyElements.filterByType[InputsSectionElement], "inputs", 1): ErrorOr[Option[InputsSectionElement]]
+      _ <- checkIfStdoutInputExist(inputValidateElement)
+      _ <- checkIfStderrInputExist(inputValidateElement)
+    } yield inputValidateElement
+
+    val intermediateValueDeclarationStdoutCheck: ErrorOr[Option[String]] = checkStdoutIntermediates(bodyElements.filterByType[IntermediateValueDeclarationElement])
+    val intermediateValueDeclarationStderrCheck: ErrorOr[Option[String]] = checkStderrIntermediates(bodyElements.filterByType[IntermediateValueDeclarationElement])
+
+    val outputsSectionValidation: ErrorOr[Option[OutputsSectionElement]] = for {
+      outputValidateElement <- validateSize(bodyElements.filterByType[OutputsSectionElement], "outputs", 1): ErrorOr[Option[OutputsSectionElement]]
+      _ <- checkIfStdoutOutputExists(outputValidateElement)
+      _ <- checkIfStderrOutputExists(outputValidateElement)
+    } yield outputValidateElement
 
     val graphSections: Vector[WorkflowGraphElement] = bodyElements.filterByType[WorkflowGraphElement]
 
     val metaSectionValidation: ErrorOr[Option[MetaSectionElement]] = validateSize(bodyElements.filterByType[MetaSectionElement], "meta", 1)
     val parameterMetaSectionValidation: ErrorOr[Option[ParameterMetaSectionElement]] = validateSize(bodyElements.filterByType[ParameterMetaSectionElement], "parameterMeta", 1)
 
-    (inputsSectionValidation, outputsSectionValidation, metaSectionValidation, parameterMetaSectionValidation) mapN {
-      (validInputs, validOutputs, meta, parameterMeta) =>
+    (inputsSectionValidation, outputsSectionValidation, metaSectionValidation, parameterMetaSectionValidation, intermediateValueDeclarationStdoutCheck, intermediateValueDeclarationStderrCheck) mapN {
+      (validInputs, validOutputs, meta, parameterMeta, _, _) =>
       WorkflowDefinitionElement(name, validInputs, graphSections.toSet, validOutputs, meta, parameterMeta, sourceLocation)
     }
   }
 
+  private def checkIfStdoutInputExist(inputSection: Option[InputsSectionElement]): ErrorOr[Option[String]] = {
```

Comment:
I only have a few minutes here before a meeting but I wanted to say this code looks good!

If there's one thing we could improve, I think we may be able to use the superclass `WorkflowBodyElement` (and equivalent superclasses for subsequent element types) to de-duplicate some of the `check...` functions.

Reply:
+1 to Adam's suggestion. I think we can also use super class `WorkflowGraphElement` which is what `IntermediateValueDeclarationElement`, `OutputDeclarationElement` and `InputDeclarationElement` extend. We can also use generics for `StdoutElement` and `StderrElement` so that we don't need two different functions for this. I was thinking of something on the lines of
```
private def check[A <: WorkflowGraphElement, B](section: Seq[A], expressionType: B, expressionName: String): ErrorOr[Option[String]] = {
    if (section.map(_.expression).exists(_.isInstanceOf[expressionType.type])) {
      s"Workflow cannot have $expressionName expression at workflow-level.".invalidNel
    } else None.validNel
  }
```
But currently it throws error saying `value expression is not a member of type parameter A` since `WorkflowGraphElement` is a very broad super class and doesn't have the parameter `expression` in its class. I don't have a fix for this yet, but I will look into this more on Monday.